src/tutorials/basics/writing-on-the-screen.md

```diff
@@ -0,0 +1,767 @@
+draw things on it. The UI is build using `app.new_ui().build().unwrap();`. In
+```rust,no_run
+# #![allow(unused_imports)]
+# #![allow(dead_code)]
+# extern crate nannou;  // tell rust that we want to use nannou
+#
+# use nannou::prelude::*;  // get a lot of convenience functions
+# use nannou::ui::prelude::*;  // get a lot of functions for UI
+# use nannou::ui::Color::Rgba;  // we want to use color later
+#
+# fn main() {
+#    nannou::app(model).update(update).run();  // run an app which can handle (time) updates
+# }
+#
+# // model of the world (on screen)
+# struct Model {
+#    ui: Ui,  // data structure to handle (bare) UI
+#    widget_ids: Ids,  // IDs of UI widgets
+#    message: String,  // message to be displayed
+#    font_id: nannou::ui::text::font::Id,  // ID of font used to display text
+# }
+#
+# // IDs of the widgets in use
+# struct Ids {
+#    text: widget::Id,  // only one text widget to be used
+# }
+#
+# fn model(app: &App) -> Model {
+#    app.new_window().event(event).view(view).build().unwrap();  // build a window for the app, handle keystrokes, draw things on it
+#    let mut ui = app.new_ui().build().unwrap(); // create a mutable UI
+#    let message = String::from(
+#        "PRESS 'L' TO LIST ALL FONT IDS\nON THE TERMINAL\n\nPRESS 'C' TO CHANGE FONT\nON SCREEN\n\nPRESS 'ESC' TO QUIT",
+#    ); // text to display
+#
+#    // generate IDs for each widget
+    let widget_ids = Ids {
+        text: ui.generate_widget_id(),
+    };
+#
+#    // load other fonts
+#    let font_map = ui.fonts_mut();  // get the mutable map of fonts for the UI
+#    let font_id = font_map.ids().next().unwrap(); // use default font ID as default
+#    let mut res = font_map.insert_from_file("assets/fonts/retroscape/Retroscape.ttf");
+#    if let Err(e) = res {
+#        println!("{}", e);
+#        panic!("loading font failed");
+#    }
+#    res = font_map.insert_from_file("assets/fonts/voynich/voynich-1.23-webfont.ttf");
+#    if let Err(e) = res {
+#        println!("{}", e);
+#        panic!("loading font failed");
+#    }
+#
+#   // generate the actual model (of the UI)
+#    Model {
+#        ui,
+#        message,
+#        font_id,
+#        widget_ids,
+#    }
+# }
+# // handle events and update the (world) model accordingly
+# fn event(_app: &App, model: &mut Model, event: WindowEvent) {}
+#
+# // handle update event to update the (world) model accordingly
+# fn update(_app: &App, model: &mut Model, _update: nannou::event::Update) {}
+#
+# // display the state of the world
+# fn view(app: &App, model: &Model, frame: Frame) -> Frame {
+#    frame
+# }
+```
+we make an `Ids` struct and generate a widget ID for the text field. `ui.generate_widget_id()`
+is necessary to gurantee unique IDs per widget so each can interact without internal confusion.
+
+Now that the UI widgets are taken care of the can handle the fonts to display the message text.
+We learned already that Nannou tracks fonts in a (global) font map. We get access to it by
```

Comment:
I think `(global)` can be removed, as global has a different meaning in Rust. The font map is owned by the `Ui` and only lives as long as the `Ui` does.